pkg/cloudprovider/cloudprovider.go

```diff
@@ -97,7 +97,8 @@ func (c *CloudProvider) Create(ctx context.Context, nodeClaim *corev1beta1.NodeC
 		if errors.IsNotFound(err) {
 			c.recorder.Publish(cloudproviderevents.NodeClaimFailedToResolveNodeClass(nodeClaim))
 		}
-		return nil, fmt.Errorf("resolving node class, %w", err)
+		// We treat a failure to resolve the NodeClass as an ICE since this means there is no capacity possibilities for this NodeClaim
+		return nil, cloudprovider.NewInsufficientCapacityError(fmt.Errorf("resolving node class, %w", err))
```

Comment:
No impact there. We don't do any updates to the ICE cache on returning this type of error. This error type is here to describe cases where you attempt to launch an instance with the Create() call but the cloudprovider comes back and tells you that there's no capacity there to do so for one reason or another. In this case, it's because we couldn't resolve the config information. 

From the NodeClaim launch side, this causes a delete of the NodeClaim and the provisioner will pick-up and try again, this time realizing that it can't use that Provisioner due to the NotFound returning no instance types